storage/aptosdb/src/lib.rs

```diff
@@ -995,24 +973,14 @@ impl DbReader for AptosDB {
     /// Get the first version that txn starts existent.
     fn get_first_txn_version(&self) -> Result<Option<Version>> {
         gauged_api("get_first_txn_version", || {
-            if let Some(pruner) = self.ledger_pruner.as_ref() {
-                // If pruning is enabled, we can get the min readable version from the pruner.
-                Ok(pruner.get_min_readable_version())
-            } else {
-                self.transaction_store.get_first_txn_version()
-            }
+            self.transaction_store.get_first_txn_version()
```

Comment:
nit: it's in theory cheaper if call `self.pruner.get_min_readable_version()` here (as well as for `get_first_write_set_version()`), any  concern doing that?

Reply:
It seems that the min_readable_version initialization logic https://github.com/aptos-labs/aptos-core/blob/38b1b4ccb2e6ca5f891c985e8424944efec62739/storage/aptosdb/src/pruner/ledger_store/ledger_store_pruner.rs#L77 (there is a default value 0) in pruners are different from the get_first_txn_version. I am not sure if we should have a default version 0.

Reply:
yeah strictly speaking when there's no genesis txn we should represent that with None :trollface: 

Is it worth it to make it right in the pruner?

Reply:
Making min_readable_version none actually will make other places quite verbose. Can we go with seeking the index for these two APIs?

Reply:
I'm fine with that.

Reply:
SG. Keep it as is.